idea/src/org/jetbrains/kotlin/idea/inspections/ConstantConditionIfInspection.kt

```diff
@@ -26,56 +26,76 @@ import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
 class ConstantConditionIfInspection : AbstractKotlinInspection() {
 
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
-        return ifExpressionVisitor(fun(expression) {
-            val condition = expression.condition ?: return
+        return ifExpressionVisitor { expression -> registerProblem(expression, holder) }
+    }
 
-            val context = condition.analyze(BodyResolveMode.PARTIAL_WITH_CFA)
-            val constantValue = condition.constantBooleanValue(context) ?: return
+    private fun registerProblem(expression: KtIfExpression, holder: ProblemsHolder?): List<ConstantConditionIfFix> {
```

Comment:
I don't like the name here (because it associates directly with IDEA `registerProblem`). Something like `getFixesAndRegisterProblem` or `collectFixesAndRegisterProblem` would be better here.